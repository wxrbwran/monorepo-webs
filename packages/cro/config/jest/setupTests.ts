// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});
