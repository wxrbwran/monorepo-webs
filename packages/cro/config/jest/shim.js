import 'core-js/stable';
import 'regenerator-runtime/runtime';

/* eslint-disable import/first */
global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function requestAnimationFrame(callback) {
    setTimeout(callback, 0);
  };

global.matchMedia = window.matchMedia = () => ({
  matches: false,
  addListener: () => { },
  removeListener: () => { },
});
