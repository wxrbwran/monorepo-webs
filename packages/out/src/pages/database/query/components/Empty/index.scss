.search_empty {
  .subtitle {
    font-size: 14px;
    color: #999999;
    .top {
      margin-bottom: 35px;
      color: #666666;
    }
  }
  .ant-result-icon {
    margin-bottom: 0;
  }
}
