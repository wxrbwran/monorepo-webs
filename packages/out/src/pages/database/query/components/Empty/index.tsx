import { Result } from 'antd';
import React from 'react';
import errorIcon from '@/assets/img/empty.png';
import './index.scss';

const Empty: React.FC<{}> = () => (
    <div className="search_empty">
      <Result
        subTitle={
          <div className="subtitle">
            <p className="top">很抱歉，没有找到相关数据。</p>
            <p>温馨提示：请检查您的查询条件是否正确</p>
            <p>如有任何意见或建议，请及时反馈给我们</p>
          </div>
        }
        icon={
          <img style={{ width: 100, height: 100 }} src={errorIcon} alt="" />
        }

      />
    </div>

);

export default Empty;
