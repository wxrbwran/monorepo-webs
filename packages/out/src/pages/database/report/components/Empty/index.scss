.report_empty {
  font-size: 14px;
  color: #666666;
  height: 100%;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  display: flex;

  .no_data_img {
    width: 100px;
    height: 100px;
  }

  p {
    text-align: center;
  }

  button {
    border-radius: 3px;
    height: 40px;
    font-size: 14px;
    margin-top: 24px;
  }
}
