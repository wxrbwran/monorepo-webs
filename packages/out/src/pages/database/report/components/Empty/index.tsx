import React from 'react';
import noTable from '@/assets/img/empty.png';
import { Link } from 'react-router-dom';
import './index.scss';

export default () => {
  return (
    <div className="report_empty">
      <img className="no_data_img" src={noTable} alt="暂无主观量表" />
      <p> 暂未生成报告， </p>
      <p>请您<Link data-testid='link' to="/database/query">去查询</Link>模块根据您的查询条件生成科研报告吧！</p>
    </div>
  )
}
