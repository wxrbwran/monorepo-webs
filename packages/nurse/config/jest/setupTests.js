// import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import 'jest';
// import axios from 'axios';
// import '@testing-library/jest-dom/extend-expect';

// Enzyme.configure({ adapter: new Adapter() });

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});

// jest.mock('@/services/api', () => {
//     const API = {
//         get: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         post: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         setAuthorizationToken: jest.fn(),
//     };
//     const http = API;
//     return http;
// })
