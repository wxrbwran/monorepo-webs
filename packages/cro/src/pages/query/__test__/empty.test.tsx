import 'jest';
import React from 'react';
import Empty from '../components/empty';
import { render, RenderResult} from '@testing-library/react';

describe('查询页面：查询结果没有找到想着数据', () => {
  let wrapper: RenderResult;
  beforeEach(() => {
    wrapper = render(<Empty />)
  })
  it('匹配快照', () => {
    expect(wrapper).toMatchSnapshot();
    // wrapper.debug();
  })
  it('ui匹配', () => {
    expect(wrapper.getByText('很抱歉，没有找到相关数据。')).toBeTruthy();
  })
})