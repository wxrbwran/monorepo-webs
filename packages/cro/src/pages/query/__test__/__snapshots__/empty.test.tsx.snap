// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`查询页面：查询结果没有找到想着数据 匹配快照 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="search-empty"
      >
        <div
          class="ant-result ant-result-info"
        >
          <div
            class="ant-result-icon"
          >
            <img
              alt=""
              src="empty.png"
              style="width: 100px; height: 100px;"
            />
          </div>
          <div
            class="ant-result-title"
          />
          <div
            class="ant-result-subtitle"
          >
            <div
              class="subtitle"
            >
              <p
                class="top"
              >
                很抱歉，没有找到相关数据。
              </p>
              <p>
                温馨提示：请检查您的查询条件是否正确
              </p>
              <p>
                如有任何意见或建议，请及时反馈给我们
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="search-empty"
    >
      <div
        class="ant-result ant-result-info"
      >
        <div
          class="ant-result-icon"
        >
          <img
            alt=""
            src="empty.png"
            style="width: 100px; height: 100px;"
          />
        </div>
        <div
          class="ant-result-title"
        />
        <div
          class="ant-result-subtitle"
        >
          <div
            class="subtitle"
          >
            <p
              class="top"
            >
              很抱歉，没有找到相关数据。
            </p>
            <p>
              温馨提示：请检查您的查询条件是否正确
            </p>
            <p>
              如有任何意见或建议，请及时反馈给我们
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
